packages/approval-controller/CHANGELOG.md

```diff
@@ -6,6 +6,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [3.5.3]
+### Changed
+- Bump dependency on `@metamask/utils` to ^8.1.0 ([#1639](https://github.com/MetaMask/core/pull/1639))
+- Move `eth-rpc-errors@^4.0.2` dependency to `@metamask/rpc-errors@^6.0.2` ([#1743](https://github.com/MetaMask/core/pull/1743))
```

Comment:
I'm currently checking whether this will cause breaking changes, stand by...

Reply:
Hmm... so it does seem this causes breaking changes. Namely, if you are still using `eth-rpc-errors` to build errors and pass them to `approval-controller`, then you need to start using `@metamask/rpc-errors` instead, otherwise you will get a type error:

```
import {ControllerMessenger} from "@metamask/base-controller";
import {ApprovalController} from "@metamask/approval-controller";
import {ethErrors} from "eth-rpc-errors";

function main() {
  const messenger = new ControllerMessenger();
  const restrictedMessenger = messenger.getRestricted({
    name: "ApprovalController",
    allowedActions: [],
    allowedEvents: [],
  });
  const approvalController = new ApprovalController({
    messenger: restrictedMessenger,
    showApprovalRequest: () => {
      // whatever
    },
    state: {},
    typesExcludedFromRateLimiting: [],
  });
  const rejectionError = ethErrors.rpc.invalidRequest({
    data: new Date()
  });
  // TypeScript error:
  // Type 'Date | undefined' is not assignable to 'DataWithOptionalCause | undefined'.
  approvalController.clear(rejectionError);
}
```

So maybe we should bump this package to 4.0.0 and leave a note here on how to upgrade?

Reply:
🤷‍♂️